dask/dataframe/io/csv.py

```diff
@@ -105,6 +105,9 @@ def __call__(self, part):
         if not is_first:
             write_header = True
             rest_kwargs.pop("skiprows", None)
+        if not is_last:
+            write_header = True
```

Comment:
I reran your original bug report and realized that if you include both skipfooter and skiprows the len of the output dataframe isn't correct. 

I think this is what is causing the issue:

```suggestion
```

Reply:
@rhjmoore  checking in here. Would you mind including this change, and merge the current main branch of dask into your branch to make sure everything is up-to-date, and will see what's the status of the tests after you push. 

Let us know if you have any troubles.

Reply:
Thanks, I think you're right and I've fixed this now. Have remerged with upstream now.